### PR TITLE
MINOR: Add ineligible replica reason to log message

### PR DESCRIPTION
**Pull request by hachikuji**
It's useful if the message about ineligible replicas added in https://github.com/apache/kafka/commit/f83d95d9a28267f7ef7a7b1e584dcdb4aa842210#diff-a52da12861f5f9722a73c354a650e4b9af3e7ccce560d7e0d1e67c299e000b58 explains the reason the replica is ineligible.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java [hunk 1]
```diff
@@ -1125,13 +1125,11 @@ private Errors validateAlterPartitionData(
             return INVALID_REQUEST;
         }
 
-        List<Integer> ineligibleReplicas = partitionData.newIsr().stream()
-            .filter(replica -> !isEligibleReplica.apply(replica))
```
Comment (dajac) [inline metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java:1129]: nit: I think that we could remove `isEligibleReplica` from the function params. It is not used anymore.

File: metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java [hunk 2]
```diff
@@ -1125,13 +1125,11 @@ private Errors validateAlterPartitionData(
...
-            .collect(Collectors.toList());
+        List<IneligibleReplica> ineligibleReplicas = ineligibleReplicasForIsr(newIsr);
         if (!ineligibleReplicas.isEmpty()) {
             log.info("Rejecting AlterPartition request from node {} for {}-{} because " +
                     "it specified ineligible replicas {} in the new ISR {}.",
-                    brokerId, topic.name, partitionId, ineligibleReplicas, partitionData.newIsr());
```
Comment (dajac) [inline metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java]: nit: We may want to revert this change in order to remain consistent with the other log line in this method.

File: metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java (hunk 1 above)
Comment (hachikuji) [reply metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java:1129]: Thanks for the reminder. Forgot about that.

**Review by dajac — APPROVED** [review]
LGTM, thanks.
